pkg/common/url_test.go

```diff
@@ -7,113 +7,90 @@ import (
 	"github.com/tj/assert"
 )
 
-func TestBuildInvokeURL(t *testing.T) {
+func TestBuildDeploymentURL(t *testing.T) {
+	externalUrl := "http://app.example.com"
+
 	tests := []struct {
 		name        string
-		externalUrl string
 		stub        *types.StubWithRelated
 		deployment  *types.Deployment
 		invokeType  string
-		want        string
+		expectedURL string
 	}{
 		{
-			name:        "should return path url for private deployments",
-			externalUrl: "http://app.example.com",
-			stub: &types.StubWithRelated{
-				Stub: types.Stub{
-					ExternalId: "c19515e6-13c7-423f-9d91-227082151538",
-					Config:     `{`,
-					Type:       types.StubType(types.StubTypeScheduledJobDeployment),
-				},
-			},
+			name: "returns host-based URL",
+			stub: &types.StubWithRelated{},
 			deployment: &types.Deployment{
-				Name:    "app1",
-				Version: uint(11),
+				Subdomain: "my-app-1234567",
+				Version:   8,
 			},
-			invokeType: InvokeUrlTypePath,
-			want:       "http://app.example.com/schedule/app1/v11",
+			invokeType:  InvokeUrlTypeHost,
+			expectedURL: "http://my-app-1234567-v8.app.example.com",
 		},
 		{
-			name:        "should return path url for public deployments",
-			externalUrl: "https://app.example.com",
-			stub: &types.StubWithRelated{
-				Stub: types.Stub{
-					ExternalId: "64741d05-2273-422a-8424-2e0760e539d3",
-					Config:     `{"authorized": false}`,
-					Type:       types.StubType(types.StubTypeEndpointDeployment),
-				},
-			},
+			name: "returns path-based URL",
+			stub: &types.StubWithRelated{Stub: types.Stub{
+				Type: types.StubType(types.StubTypeEndpointDeployment),
+			}},
 			deployment: &types.Deployment{
-				Name:    "app2",
-				Version: uint(22),
+				Name:    "my-app",
+				Version: 8,
 			},
-			invokeType: InvokeUrlTypePath,
-			want:       "https://app.example.com/endpoint/public/64741d05-2273-422a-8424-2e0760e539d3",
+			invokeType:  InvokeUrlTypePath,
+			expectedURL: "http://app.example.com/endpoint/my-app/v8",
 		},
 		{
-			name:        "should return path url for serves",
-			externalUrl: "https://app.example.com",
-			stub: &types.StubWithRelated{
-				Stub: types.Stub{
-					ExternalId: "aff86f02-c968-47a9-9132-0bde826b0aca",
-					Config:     `{}`,
-					Type:       types.StubType(types.StubTypeASGIServe),
-				},
-			},
-			deployment: &types.Deployment{
-				Name:    "app2",
-				Version: uint(28),
-			},
-			invokeType: InvokeUrlTypePath,
-			want:       "https://app.example.com/asgi/id/aff86f02-c968-47a9-9132-0bde826b0aca",
+			name: "returns public path-based URL",
+			stub: &types.StubWithRelated{Stub: types.Stub{
+				Type:       types.StubType(types.StubTypeEndpointDeployment),
+				Config:     `{"authorized": false}`,
+				ExternalId: "e9c29586-c465-4a67-9c9b-25293d1ce77b",
+			}},
+			invokeType:  InvokeUrlTypePath,
+			expectedURL: "http://app.example.com/endpoint/public/e9c29586-c465-4a67-9c9b-25293d1ce77b",
```

Comment:
are there public subdomain-based urls?

Reply:
I feel like we talked about this but I forgot the reason why we can't

Reply:
There used to be, but @mernit and I talked about it and we thought it made sense to just have the host-based URL be the same for public and private endpoints. This keeps to what v1 was doing IIRC.

For example, http://my-app-1234567-v8.app.example.com will be the same whether `authorized` is true or false. We also won't print out the auth header if `authorized` is false, same as before.

Reply:
ok so http://my-app-1234567-v8.app.example.com/ can still be seen as a public endpoint

Reply:
Yes, public and private (authorized/unauthorized) endpoints will have the same host-based URL.